cpp/celeborn/network/MessageDecoder.h

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <folly/io/Cursor.h>
+#include <wangle/codec/ByteToMessageDecoder.h>
+
+namespace celeborn {
+namespace network {
+/**
+ * A complete Message encoding/decoding frame is:
+ * -----------------------------------------------------------------------
+ * | encodedLength | msgType | bodyLength | encodedContent | bodyContent |
+ * -----------------------------------------------------------------------
+ * The size of each part is:
+ * -----------------------------------------------------------------------
+ * | 4             | 1       | 4          | #encodedLength | #bodyLength |
+ * -----------------------------------------------------------------------
+ * So the #headerLength is 4 + 1 + 4,
+ * and the complete frameLength is:
+ *   #frameLength = #headerLength + #encodedLength + #bodyLength.
+ */
+
+class MessageDecoder : public wangle::ByteToByteDecoder {
```

Comment:
It is a frame decoder, not a message decoder. How about changing its name to avoid name conflicts?